internal/controller/githubissue_controller_test.go

```diff
@@ -0,0 +1,149 @@
+	"net/http"
+
+	"github.com/migueleliasweb/go-github-mock/src/mock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"math/rand"
+	"time"
+
+	issuesv1alpha1 "github.com/matanamar10/github-issue-operator-hhome-assignment/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func RandomString() string {
+	source := rand.NewSource(time.Now().UnixNano())
+	rng := rand.New(source)
+
+	length := 8
+	charset := "abcdefghijklmnopqrstuvwxyz0123456789"
+
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[rng.Intn(len(charset))]
+	}
+	return string(b)
+}
+
+func GenerateTestIssue() *issuesv1alpha1.GithubIssue {
+	name := RandomString()
+	title := RandomString()
+	description := RandomString()
+	newIssue := &issuesv1alpha1.GithubIssue{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "default",
+		},
+		Spec: issuesv1alpha1.GithubIssueSpec{
+			Title:       title,
+			Description: description,
+			Repo:        "https://github.com/test/test",
+		},
+	}
+	return newIssue
+}
+
+var (
+	timeout  = time.Second * 20
+	interval = time.Millisecond * 250
+)
+
+var _ = Describe("githubIssue controller e2e test", func() {
+	Context("e2e testing", func() {
+		It("creates an issue", func() {
+			name := fmt.Sprintf("e2e-test-%s", RandomString())
+			testIssue := &issuesv1alpha1.GithubIssue{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: "default",
+				},
+				Spec: issuesv1alpha1.GithubIssueSpec{
+					Title:       name,
+					Description: "this is generated from an e2e-test",
+					Repo:        "https://github.com/matanamar10/github-issue-operator-home-assignment",
+				},
+			}
+			err := k8sClient.Create(ctx, testIssue)
+			Expect(err).ToNot(HaveOccurred())
+
+			githubIssueReconciled := issuesv1alpha1.GithubIssue{}
+			req := types.NamespacedName{
+				Name:      testIssue.ObjectMeta.Name,
+				Namespace: testIssue.Namespace,
+			}
+
+			Eventually(func() bool {
+				Expect(k8sClient.Get(ctx, req, &githubIssueReconciled)).Should(BeNil(), "should find resource")
+				return meta.IsStatusConditionTrue(githubIssueReconciled.Status.Conditions, "IssueIsOpen")
+			}, timeout, interval).Should(BeTrue())
+
+			By("updating issue")
+			githubIssueReconciled.Spec.Description = "updated description"
+			Expect(k8sClient.Update(ctx, &githubIssueReconciled)).Should(Succeed())
+
+			By("deleting issue")
+			Expect(k8sClient.Delete(ctx, &githubIssueReconciled)).Should(Succeed())
+
+			deletedIssue := &issuesv1alpha1.GithubIssue{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, req, deletedIssue)
+				return k8serrors.IsNotFound(err)
+			}, timeout, interval).Should(BeTrue())
+		})
+	})
+})
+
+var _ = Describe("githubIssue controller", func() {
+	Context("When creating githubIssue", func() {
+		It("Receive error when trying to create an issue", func() {
```

Comment:
Where are the rest of the test cases? Your code does more than just creating an issue, so test it...